src/lib/Components/FilterModal.tsx

```diff
@@ -1,169 +1,141 @@
+import { ArtworkFilterContext } from "../utils/ArtworkFiltersStore"
+import { SortOptionsScreen as SortOptions } from "./ArtworkFilterOptions/SortOptions"
 
 interface FilterModalProps extends ViewProperties {
   closeModal?: () => void
   navigator?: NavigatorIOS
   isFilterArtworksModalVisible: boolean
 }
 
-interface FilterModalState {
-  isComponentMounted: boolean
-  sortableItems: Array<{ type: string; data: any }>
-}
+export const FilterModalNavigator: React.SFC<FilterModalProps> = ({ closeModal, isFilterArtworksModalVisible }) => {
+  const { dispatch, state } = useContext(ArtworkFilterContext)
 
-export class FilterModalNavigator extends React.Component<FilterModalProps, FilterModalState> {
-  state: FilterModalState = {
-    isComponentMounted: false,
-    sortableItems: [],
+  const handleClosingModal = () => {
+    closeModal()
+    dispatch({ type: "resetFilters" })
   }
 
-  componentDidMount() {
-    setTimeout(() => {
-      LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
-      this.setState({ isComponentMounted: true })
-    }, 100)
+  const applyFilters = () => {
+    dispatch({ type: "applyFilters", payload: [{ type: state.selectedFilters.type, filter: "sort" }] })
+    closeModal()
   }
 
-  closeModal() {
-    if (this.props.closeModal) {
-      this.props.closeModal()
-    }
-  }
+  const getApplyButtonCount = () => {
+    const selectedFiltersSum = state.selectedFilters.length
 
-  render() {
-    const { isFilterArtworksModalVisible } = this.props
-    const { isComponentMounted } = this.state
-
-    return (
-      <>
-        {isFilterArtworksModalVisible && (
-          <RNModal animationType="fade" transparent={true} visible={isFilterArtworksModalVisible}>
-            <TouchableWithoutFeedback onPress={null}>
-              <ModalBackgroundView>
-                <TouchableOpacity onPress={this.props.closeModal} style={{ flexGrow: 1 }} />
-                <ModalInnerView visible={isComponentMounted}>
-                  <NavigatorIOS
-                    navigationBarHidden={true}
-                    initialRoute={{
-                      component: FilterOptions,
-                      passProps: { closeModal: this.props.closeModal },
-                      title: "", // this property (can be an empty string) is required otherwise RN throws a warning
-                    }}
-                    style={{ flex: 1 }}
-                  />
-                  <Box p={2}>
-                    <Button onPress={() => this.closeModal()} block width={100} variant="secondaryOutline">
-                      Apply
-                    </Button>
-                  </Box>
-                </ModalInnerView>
-              </ModalBackgroundView>
-            </TouchableWithoutFeedback>
-          </RNModal>
-        )}
-      </>
-    )
+    return selectedFiltersSum > 0 ? "Apply" + " (" + selectedFiltersSum + ")" : "Apply"
```

Comment:
#nonblocking This might be a bit more readable with backticks

```suggestion
    return selectedFiltersSum > 0 ? `Apply (${selectedFiltersSum})` : "Apply"
```